src/main/java/tool/TrivyGrypeOutputProcessor.java

```diff
@@ -0,0 +1,213 @@
+//                        for (String key : JSONObject.getNames(cvssData)) {
+//                            JSONObject cvss = cvssData.optJSONObject(key);
+//                            if (cvss != null && cvss.has("V3Score")) {
+//                                findingSeverity =  cvss.optString("V3Score");
+//                            }
+//                        }
+//                        if (findingSeverity.equals("")){
+//                            findingSeverity = "0.0";
+//                        }
+//                    }
+                }
+                String vulnId = formatVulnerabilityId(rawId);
+                ArrayList<String> cwes = vulnId.contains("GHSA") ? retrieveGhsaCwes(vulnId) : retrieveNvdCwes(vulnId);
+                if (!cwes.isEmpty()) {
+                    toolVulnerabilities.add(new RelevantVulnerabilityData(vulnId, cwes, helperFunctions.severityToInt(findingSeverity)));
+                }
+            }
+        } catch (JSONException e) {
+            // This intentionally lacks a throw.
+            // No json result is a valid program state and is handled elsewhere
+            LOGGER.warn("Unable to parse json. ", e);
+        }
+
+        return toolVulnerabilities;
+    }
+
+    /**
+     * Builds Finding and Diagnostic objects from the list of ToolVulnerabilities generated previously.
+     * Adds the new Diagnostics to the PIQUE tree.
+     *
+     * @param toolVulnerabilities ArrayList of RelevantVulnerabilityData objects representing Output from Tool
+     * @param diagnostics Map of diagnostics for Tool output
+     */
+    // TODO This void method is a little confusing - improve?
+    @Override
+    public void addDiagnostics(ArrayList<RelevantVulnerabilityData> toolVulnerabilities, Map<String, Diagnostic> diagnostics, String toolName) {
+        for (RelevantVulnerabilityData relevantVulnerabilityData : toolVulnerabilities) {
+            Diagnostic diag = diagnostics.get(relevantVulnerabilityData.getCwe().get(0) + toolName);
+            if (diag == null) {
+                diag = diagnostics.get("CWE-other" + toolName);
+                LOGGER.warn("CVE with CWE outside of CWE-699 found.");
+            }
+            Finding finding = new Finding("", 0, 0, relevantVulnerabilityData.getSeverity());
+            finding.setName(relevantVulnerabilityData.getCve());
+            diag.setChild(finding);
+        }
+    }
+
+    /**
+     * Retrieves CWE list for given CVE from the GitHub Vulnerabiity Database
+     *
+     * @param cveId one cve identified in the tool output
+     * @return all associated CWEs for the cveId
+     */
+    private ArrayList<String> retrieveGhsaCwes(String cveId) {
+        // TODO How many GHSAs are likely? Rate limit issues? Batch API calls?
+        ArrayList<String> cwes = new ArrayList<>();
+
+        Properties prop = PiqueProperties.getProperties();	// TODO this might already been injected at the class level
```

Comment:
Can probably remove this TODO comment.  (I know I put it in, but if this is the only place we need to getProperties(), then the local instantiation is the right way to go)